### DISCUSSION TITLE
Script to right-align text quickly?

### DISCUSSION THREAD

**Discussion by daamiian** [SilentVoid13/Templater#672] [Help] [answered]
I'm new to Templater, I'd really appreciate some help figuring out how to create a script that takes my selected text and right aligns it.

The plan was to use `<p style='text-align: right;'> TEXT HERE </p>` to actually right align my text in Obsidian
then to automate `tp.file.cursor()` to the TEXT HERE position in the above line
then `tp.file.selection()` to return the selected text from the file to be templated which I want to right-align in the cursor position.

Please, any help is appreciated.

**Comment by welpdx** [comment, marked as answer]
Here you go:
```js
<%*
if (tp.file.selection() !== "" ) {
	tR+= `<p style='text-align: right;'>` + tp.file.selection() + tp.file.cursor() +`</p>`
}
%>
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/kNJtr77.gif" alt="demo" />
</details>

I actually use this quite frequently to highlight my notes! For example, to highlight a text orange
```js
<%*
if (tp.file.selection() !== "" ) {
	tR+= `<span style='background-color: Orange;color:black'>` + tp.file.selection() + tp.file.cursor() +`</span>`
}
%>
```

<details>
           <summary style="font-size:20">Demo text highlighter</summary>
           <img src="https://i.imgur.com/Ezh3GPm.gif" alt="demo" />
</details>


Hope that helps.

~welp

**Reply by daamiian** [reply]
@welpdx thank you so much. Do you know of any good resources where I can learn the syntax for this? I never really programmed, only touching some python and matlab in university. I've since been coding decommissioned but the basics are still in my mind. For example I don't know why you used += for the tR variable. Thanks again for the reply, you saved me a lot of time!

**Reply by welpdx** [reply]
> thank you so much

hey no problem!

> only touching some python and matlab in university

ah lemme guess... you are in engineering? XD

> know of any good resources where I can learn the syntax for this

So Templater uses a plugin called [Eta](https://github.com/eta-dev/eta), the syntax is mentioned [here](https://silentvoid13.github.io/Templater/syntax.html) in the [Templater Documentation](https://silentvoid13.github.io/Templater/). Templater also has some cool [internal functions, see here](https://silentvoid13.github.io/Templater/internal-functions/overview.html).

I recommend you just play around with it. I am not a coder either, but I find that playing around helps me learn. There are alot of good [templater scripts in the show case here](https://github.com/SilentVoid13/Templater/discussions/categories/templates-showcase). 


If you are unfamiliar with javascript, it might be a good idea to learn the fundamentals. Like install vscode and open up [some beginner js course on yt](https://www.youtube.com/watch?v=PkZNo7MFNFg), and just have some fun! XD

Obsidian is like chrome basically and by learning javascript and perhaps some html, you can really tinker with it for your own workflow and at the same time, you are doing front end development!

gl hf!

(and if I have answered your original question, could you please mark this post as answered? thanks)

**Reply by daamiian** [reply]
> ah lemme guess... you are in engineering? XD

got me xD

Thanks for all of the help. I'll take a look at the beginner courses for sure.
